src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -87,8 +103,138 @@ internal IEnumerable<NodeSearchElement> GetMatchingNodes()
                 if (element.OutputParameters.Any(op => op == inputPortType))
                     elements.Add(element);
             }
+            elements.Sort(CompareNodeSearchElementByTypeDistance);
+            
+            return elements.OrderBy(x =>x.Group);
+        }
+
+        /// <summary>
+        /// Does typeb derive from typea
+        /// </summary>
+        /// <param name="typea"></param>
+        /// <param name="typeb"></param>
+        /// <param name="core"></param>
+        /// <returns></returns>
+        private bool DerivesFrom(string typea, string typeb, ProtoCore.Core core)
+        {
+            try
+            {
+                //TODO mirrors can be cached until new types are imported...
+                var mirror1 = new ClassMirror(typea, core);
+                var mirror2 = new ClassMirror(typeb, core);
 
-            return elements;
+
+                //TODO as we do this check we can cache the type distance...
+                if (mirror2.GetClassHierarchy().Any(x => x.ClassName == mirror1.ClassName))
+                {
+                    //this is a derived type
+                    return true;
+                }
+                return false;
+            }
+            catch
+            {
+                // TODO need to deal with the fact that type[] and type[]..[] are not classes - trying to create these up as ClassMirrors will yield an exception.
+                Console.WriteLine($"failed to create class mirror for {typea} or {typeb} during node autocomplete operation ");
+                return false;
+            }
         }
+
+
+        // TODO should we sort at the functiondescriptor stage?
+        private int CompareNodeSearchElementByTypeDistance(NodeSearchElement x, NodeSearchElement y)
```

Comment:
@mjkkirschner Notice all of these are done at the click time, team also talked about doing these `grooming` at Dynamo startup level so we offset these operations. An interesting scale problem IMHO.